### ISSUE TITLE
Loss and grad norm explode while running pretrain

### ISSUE THREAD

**Issue by moewiee** [canopyai/Orpheus-TTS#78]
![Image](https://github.com/user-attachments/assets/516663e2-1c55-4961-9d0f-0e8fc5b0760f)

I tried to re-produce the pretraining process with Japanese dataset, however I am experiencing weird loss and grad norm logs. Things are fine at first, then after several hundred of iters, loss spike while grad norm increase gradually. Anyone having same experience, or it's just me?

Settings:

- Japanese Instruct LLM dataset, 9m rows
- Japanese TTS dataset, filter sample with SNAC-reconstructed DNSMOS > 2.5, 15m rows
- LR 5e-5
- BS 2
- BF16

**Comment by xjohnxjohn** [comment]
Hi, what the datasets you use to pretrain? Thank you

**Comment by moewiee** [comment]
@xjohnxjohn 
https://huggingface.co/datasets/izumi-lab/llm-japanese-dataset for LLM and jchat for TTS

**Comment by amuvarma13** [comment]
Hey there - yes we occasionally experience this issue - here are a few things that may be useful to try:

- Make sure you use bf16 not fp16
- Mess around with learning rate (add warmup)
- Decrease max length to 1024/2048
- Increase batch size 

Let me know if any of these help if not I can try to help debug if I see your script/config.

**Comment by moewiee** [comment]
@amuvarma13 Yeah we already using bf16 for training, decrease input max length to 2k in collate function, and increase batch size to 64 (8 gpus, 2 sample each card, 4 accumulation steps, not sure if accumulation cause any problem as Unsloth once pointed it out https://unsloth.ai/blog/gradient).

This time decrease the LR from 5e-5 to 6e-5, log seem nicer.

![Image](https://github.com/user-attachments/assets/6e072d14-2a05-40e4-83a3-57d501e9fead)

**Comment by moewiee** [comment]
@amuvarma13 currently I dont see any big spike in loss anymore. Will run several more experiment and report if run into this issue more. 
But for now, I think the issues is fixed by decreasing LR and add 10k step warm up. (bf16, max length 2k and bs 64 alone did not fix this issue, but reasonable LR + warm up seems did)

**Comment by amuvarma13** [comment]
Okay sounds good - I'm going to close this issue for now - feel free to recomment here if you run into further/the same problems.
